homeassistant/components/roborock/switch.py

```diff
@@ -120,114 +95,74 @@ async def async_setup_entry(
-
     coordinators: dict[str, RoborockDataUpdateCoordinator] = hass.data[DOMAIN][
         config_entry.entry_id
     ]
     possible_entities: list[
-        tuple[str, RoborockDataUpdateCoordinator, RoborockSwitchDescription]
+        tuple[RoborockDataUpdateCoordinator, RoborockSwitchDescription]
     ] = [
-        (device_id, coordinator, description)
-        for device_id, coordinator in coordinators.items()
+        (coordinator, description)
+        for coordinator in coordinators.values()
         for description in SWITCH_DESCRIPTIONS
     ]
     # We need to check if this function is supported by the device.
     results = await asyncio.gather(
         *(
-            description.check_support(coordinator)
-            for _, coordinator, description in possible_entities
+            coordinator.api.cache.get(description.cache_key).async_value()
+            for coordinator, description in possible_entities
         ),
         return_exceptions=True,
     )
-    valid_entities: list[RoborockNonCoordinatedSwitchEntity] = []
-    for posible_entity, result in zip(possible_entities, results):
-        if isinstance(result, Exception):
-            if not isinstance(result, RoborockException):
-                raise result
+    valid_entities: list[RoborockSwitch] = []
+    for (coordinator, description), result in zip(possible_entities, results):
+        if result is None or isinstance(result, Exception):
             _LOGGER.debug("Not adding entity because of %s", result)
         else:
             valid_entities.append(
-                RoborockNonCoordinatedSwitchEntity(
-                    f"{posible_entity[2].key}_{slugify(posible_entity[0])}",
-                    posible_entity[1],
-                    posible_entity[2],
-                    result,
+                RoborockSwitch(
+                    f"{description.key}_{slugify(coordinator.roborock_device_info.device.duid)}",
+                    coordinator.device_info,
+                    description,
+                    coordinator.api,
                 )
             )
-    async_add_entities(
-        valid_entities,
-        True,
-    )
-    async_add_entities(
-        (
-            RoborockCoordinatedSwitchEntity(
-                f"{description.key}_{slugify(device_id)}",
-                coordinator,
-                description,
-            )
-            for device_id, coordinator in coordinators.items()
-            for description in COORDINATED_SWITCH_DESCRIPTION
-            if description.check_support(coordinator) is not None
-        )
-    )
+    async_add_entities(valid_entities)
 
 
-class RoborockNonCoordinatedSwitchEntity(RoborockEntity, SwitchEntity):
-    """A class to let you turn functionality on Roborock devices on and off that does not need a coordinator."""
+class RoborockSwitch(RoborockEntity, SwitchEntity):
+    """A class to let you turn functionality on Roborock devices on and off that does need a coordinator."""
```

Comment:
Please break docstrings at 72 characters per line. The first line must be a single sentence and is a header line. After that comes a blank line and then än optional body.